edflow/iterators/tf_evaluator.py

```diff
@@ -83,3 +89,55 @@ def __init__(self, *args, **kwargs):
 
     def step_ops(self):
         return self.model.outputs
+
+
+class TFBaseEvaluator2(TFHookedModelIterator):
+    def __init__(
+        self,
+        *args,
+        checkpoint_path=None,
+        desc="Eval",
+        hook_freq=1,
+        num_epochs=1,
+        **kwargs
+    ):
+        """
+        New Base evaluator restores given checkpoint path if provided,
+        else scans checkpoint directory for latest checkpoint and uses that
+
+        Parameters
+        ----------
+        desc : str
+            a description for the evaluator. This description will be used during the logging.
+        hook_freq : int
+            Frequency at which hooks are evaluated.
+        num_epochs : int
+            Number of times to iterate over the data.
+        """
+        kwargs.update({"desc": desc, "hook_freq": hook_freq, "num_epochs": num_epochs})
+        super().__init__(*args, **kwargs)
+
+        # wait for new checkpoint and restore
+        self.restore_variables = self.model.variables
+        if checkpoint_path:
+            restorer = RestoreCurrentCheckpointHook(
+                variables=self.restore_variables,
+                checkpoint_path=checkpoint_path,
+                global_step_setter=self.set_global_step,
+            )
+            self.hooks += [restorer]
+        else:
+            restorer = RestoreTFModelHook(
+                variables=self.restore_variables,
+                checkpoint_path=ProjectManager.checkpoints,
+                global_step_setter=self.set_global_step,
+            )
+            waiter = WaitForCheckpointHook(
+                checkpoint_root=ProjectManager.checkpoints,
+                callback=restorer,
+                eval_all=self.config.get("eval_all", False),
+            )
+            self.hooks += [waiter]
```

Comment:
see comment below. We should move this logic to the `initialize` method. Also, can't we do it on `TFBaseEvaluator` itself, instead of introducing a second class? This also seems perfectly backward compatible because of the `if checkpoint_path` test.

Reply:
should be fixed